src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -40,16 +40,18 @@ class_exists(ServerBag::class);
  */
 class Request
 {
-    const HEADER_FORWARDED = 0b000001; // When using RFC 7239
-    const HEADER_X_FORWARDED_FOR = 0b000010;
-    const HEADER_X_FORWARDED_HOST = 0b000100;
-    const HEADER_X_FORWARDED_PROTO = 0b001000;
-    const HEADER_X_FORWARDED_PORT = 0b010000;
-    const HEADER_X_FORWARDED_PREFIX = 0b100000;
-
-    const HEADER_X_FORWARDED_ALL = 0b011110; // All "X-Forwarded-*" headers sent by "usual" reverse proxy
-    const HEADER_X_FORWARDED_AWS_ELB = 0b011010; // AWS ELB doesn't send X-Forwarded-Host
-    const HEADER_X_FORWARDED_TRAEFIK = 0b111110; // All "X-Forwarded-*" headers sent by Traefik reverse proxy
+    const HEADER_FORWARDED = 0b0000001; // When using RFC 7239
+    const HEADER_X_FORWARDED_FOR = 0b0000010;
+    const HEADER_X_FORWARDED_HOST = 0b0000100;
+    const HEADER_X_FORWARDED_PROTO = 0b0001000;
+    const HEADER_X_FORWARDED_PORT = 0b0010000;
+    const HEADER_X_FORWARDED_PREFIX = 0b0100000;
+    private const CONST_DEPRECATED = 0b10000000; // reserved to deprecate constants
```

Comment:
unused?

Reply:
No, the constant is not use itself, but the 8th bit is used by `HEADER_X_FORWARDED_ALL` to distinguish this exact constant from `HEADER_X_FORWARDED_FOR |  ..HOST | ..PORT | ..PROTO` and to trigger the deprecation.

This constant is a placeholder to avoid a future `HEADER_X_FORWARDED_FOOBAR = 0b10000000;` that would trigger the deprecation when people use `HEADER_X_FORWARDED_FOOBAR | ..FOR |  ..HOST | ..PORT | ..PROTO`.

Reply:
instead of a dummy const, let's add a test case that documents this?

Reply:
Done, by checking value of constants exosed by the class